packages/stable/src/api/geospatial/featureAPI.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  BaseResourceAPI,
+  CogniteExternalId,
+  ExternalId,
+} from '@cognite/sdk-core';
+import {
+  Feature,
+  FeatureAggregateParams,
+  FeatureCreateItem,
+  FeatureOutputParams,
+  FeatureSearchFilter,
+  FeatureSearchStreamFilter,
+} from './types';
+
+export class FeatureAPI extends BaseResourceAPI<Feature> {
+  public create = (
+    featureTypeExternalId: CogniteExternalId,
+    features: FeatureCreateItem[]
+  ): Promise<Feature[]> => {
+    return this.createEndpoint<FeatureCreateItem>(
+      features,
+      this.url(`${featureTypeExternalId}/features`)
+    );
+  };
+
+  public retrieve = (
+    featureTypeExternalId: CogniteExternalId,
+    externalIds: ExternalId[],
+    params?: FeatureOutputParams
+  ): Promise<Feature[]> => {
+    return this.callEndpointWithMergeAndTransform(externalIds, request =>
+      this.postInParallelWithAutomaticChunking({
+        path: this.url(`${featureTypeExternalId}/features/byids`),
+        items: request,
+        params,
+      })
+    );
+  };
+
+  public update = (
+    featureTypeExternalId: CogniteExternalId,
+    changes: FeatureCreateItem[]
+  ): Promise<Feature[]> => {
+    return this.updateEndpoint(
+      changes,
+      this.url(`${featureTypeExternalId}/features/update`)
+    );
+  };
+
+  public delete = (
+    featureTypeExternalId: CogniteExternalId,
+    externalIds: ExternalId[],
+    params: FeatureOutputParams = {}
+  ) => {
+    return this.deleteEndpoint(
+      externalIds,
+      params,
+      this.url(`${featureTypeExternalId}/features/delete`)
+    );
+  };
+
+  public search = (
+    featureTypeExternalId: CogniteExternalId,
+    params: FeatureSearchFilter = {}
```

Comment:
I would suggest to rename `params` to `filter` here and also in other methods.

Reply:
but filter is a parameter inside params 🤔